test/profiling/MemoryTestSubscriber.cpp

```diff
@@ -78,244 +64,272 @@ bool MemoryTestSubscriber::init(
     m_echo = echo;
     n_samples = nsam;
     m_data_size = data_size;
-    dynamic_data = dynamic_types;
+    dynamic_data_ = dynamic_types;
 
-    if (dynamic_data) // Dummy type registration
+    if (dynamic_data_) // Dummy type registration
     {
         // Create basic builders
-        DynamicTypeBuilder_ptr struct_type_builder(DynamicTypeBuilderFactory::get_instance()->create_struct_builder());
+        DynamicTypeBuilderFactory::_ref_type factory {DynamicTypeBuilderFactory::get_instance()};
 
-        // Add members to the struct.
-        struct_type_builder->add_member(0, "seqnum", DynamicTypeBuilderFactory::get_instance()->create_uint32_type());
-        struct_type_builder->add_member(1, "data",
-                DynamicTypeBuilderFactory::get_instance()->create_sequence_builder(
-                    DynamicTypeBuilderFactory::get_instance()->create_byte_type(), ::dds::core::LENGTH_UNLIMITED
-                    ));
-        struct_type_builder->set_name("MemoryType");
+        // Create basic builders
+        TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+        type_descriptor->kind(TK_STRUCTURE);
+        type_descriptor->name("MemoryType");
+        DynamicTypeBuilder::_ref_type struct_type_builder(factory->create_type(type_descriptor));
 
+        // Add members to the struct.
+        MemberDescriptor::_ref_type member_descriptor {traits<MemberDescriptor>::make_shared()};
+        member_descriptor->type(factory->get_primitive_type(eprosima::fastdds::dds::TK_UINT32));
+        member_descriptor->name("seqnum");
+        struct_type_builder->add_member(member_descriptor);
+        member_descriptor = traits<MemberDescriptor>::make_shared();
+        member_descriptor->type(factory->create_sequence_type(factory->get_primitive_type(eprosima::fastdds::dds::
+                        TK_UINT32), 0)->build());
```

Comment:
Same suggestion as in MemoryTestsPublisher

Reply:
Done in d5af16ae0